allennlp/nn/beam_search.py

```diff
@@ -489,7 +528,35 @@ def top_p_sampling(
             sampler=sampler_p,
         )
 
+    @classmethod
+    def stochastic_beam_search(
+        cls,
+        end_index: int,
+        max_steps: int = 50,
+        beam_size: int = 10,
+        per_node_beam_size: int = None,
+        temperature: float = 1.0,
+    ) -> "BeamSearch":
+        """
+        Given an index of the end token in target vocabulary, return a `BeamSearch` object
+        that can be used to find `beam_size` candidate sequences, found by using Stochastic
+        Beam Search, which leverages Gumbel-TopK sampling. See
+        [*Stochastic Beams and Where to Find Them: The Gumbel-Top-k Trick for Sampling
+        Sequences Without Replacement*, W Kool, H Van Hoof and M Welling, 2010]
+        (https://arxiv.org/abs/1903.06059).
```

Comment:
I don't think this will render properly in the documentation. It might have to look like:

```suggestion
        [*Stochastic Beams and Where to Find Them: The Gumbel-Top-k Trick for Sampling Sequences Without Replacement*, W Kool, H Van Hoof and M Welling, 2010]
        (https://arxiv.org/abs/1903.06059).
```

You can check this by building and serving the docs locally with `make serve-docs` and then opening [http://127.0.0.1:8000/api/nn/beam_search/](http://127.0.0.1:8000/api/nn/beam_search/) in your browser.

Reply:
This looks fine when viewing the docs, and the linter complains otherwise so I left it!